base/vault.yaml

```diff
@@ -140,6 +147,84 @@ spec:
             - name: nginx-config
               mountPath: /etc/nginx/conf.d
       containers:
+        - name: initializer
+          image: alpine
+          command:
+            - /bin/sh
+            - -c
+            - |
+              # Only run the initializer in the first replica
+              replica="${HOSTNAME: -1}";
+              if [ "${replica}" != "0" ];then
+                echo "this replica is not responsible for initialization, going to sleep";
+                while true; do sleep 86400; done
+              fi
+
+              # Validations and defaults
+              : ${VAULT_CACERT:?"Need to set VAULT_CACERT"};
+              : ${KUBECTL_VERSION:?"Need to set KUBECTL_VERSION"};
+              apk -q add curl jq;
+              vault_name="${VAULT_ADDR:-"vault"}";
+              leader_addr="https://${vault_name}-0.${vault_name}:8200";
+              replicas="${VAULT_REPLICAS:-"3"}";
+
+              # Wait for vault api and check if initialization is needed
+              until curl -s --cacert "${VAULT_CACERT}" "${leader_addr}/v1/sys/init"; do
+                echo 'leader not ready, sleeping for 3 seconds';
+                sleep 3;
+              done;
+              initialized=$(curl -s --cacert "${VAULT_CACERT}" "${leader_addr}/v1/sys/init" | jq '.initialized');
+              if [ "${initialized}" = "true" ];then
+                echo "vault is already initialized, going to sleep";
+                while true; do sleep 86400; done
+              fi
+
+              # Install kubectl
+              curl -sLO "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl";
+              chmod +x ./kubectl;
+              mv ./kubectl /usr/local/bin;
```

Comment:
You could probably achieve your kube api calls with curl instead of kubectl. That means one less thing to download and you aren't tied down to a specific version of kubectl. Depends how much of a purist you are.

Reply:
Its definitely tempting, having one less place to upgrade.

Reply:
I went for ease of use and shorter script. Since this only happens once in the sts lifetime, I think it's ok. But getting rid of the version is a big think, I'll have a look

Reply:
To that effect you might just want to get "latest" and hope the commands you are running are simple enough they are unlikely to break between versions.

Reply:
Updated to defaulting to latest but allowing setting the version if needed